Mathlib/CategoryTheory/Adjunction/Additive.lean

```diff
@@ -0,0 +1,154 @@
+
+variable [F.Additive]
+
+/-- If we have an adjunction `adj : F ⊣ G` of functors between preadditive categories,
+and if `F` is additive, then the hom set equivalence upgrades to an `AddEquiv`.
+Note that `F` is additive if and only if `G` is, by `Adjunction.right_adjoint_additive` and
+`Adjunction.left_adjoint_additive`.
+-/
+def homAddEquiv (X : C) (Y : D) : AddEquiv (F.obj X ⟶ Y) (X ⟶ G.obj Y) :=
+  {
+    adj.homEquiv _ _ with
+    map_add' _ _ := by
+      have := adj.right_adjoint_additive
+      simp [homEquiv_apply] }
+
+@[simp]
+lemma homAddEquiv_apply (X : C) (Y : D) (f : F.obj X ⟶ Y) :
+    adj.homAddEquiv X Y f = adj.homEquiv X Y f := rfl
+
+@[simp]
+lemma homAddEquiv_symm_apply (X : C) (Y : D) (f : X ⟶ G.obj Y) :
+    (adj.homAddEquiv X Y).symm f = (adj.homEquiv X Y).symm f := rfl
+
+@[simp]
+lemma homAddEquiv_zero (X : C) (Y : D) : adj.homEquiv X Y 0 = 0 := by
+  change adj.homAddEquiv X Y 0 = 0
+  rw [map_zero]
+
+@[simp]
+lemma homAddEquiv_add (X : C) (Y : D) (f f' : F.obj X ⟶ Y) :
+    adj.homEquiv X Y (f + f') = adj.homEquiv X Y f + adj.homEquiv X Y f' := by
+  change adj.homAddEquiv X Y (f + f') = _
+  simp [AddEquivClass.map_add]
+
+@[simp]
+lemma homAddEquiv_sub (X : C) (Y : D) (f f' : F.obj X ⟶ Y) :
+    adj.homEquiv X Y (f - f') = adj.homEquiv X Y f - adj.homEquiv X Y f' := by
+  change adj.homAddEquiv X Y (f - f') = _
+  simp [AddEquiv.map_sub]
+
+@[simp]
+lemma homAddEquiv_neg (X : C) (Y : D) (f : F.obj X ⟶ Y) :
+    adj.homEquiv X Y (- f) = - adj.homEquiv X Y f := by
+  change adj.homAddEquiv X Y (- f) = _
+  simp [AddEquiv.map_neg]
+
+@[simp]
+lemma homAddEquiv_symm_zero (X : C) (Y : D) :
+    (adj.homEquiv X Y).symm 0 = 0 := by
+  change (adj.homAddEquiv X Y).symm 0 = 0
+  rw [map_zero]
+
+@[simp]
+lemma homAddEquiv_symm_add (X : C) (Y : D) (f f' : X ⟶ G.obj Y) :
+    (adj.homEquiv X Y).symm (f + f') = (adj.homEquiv X Y).symm f + (adj.homEquiv X Y).symm f' := by
+  change (adj.homAddEquiv X Y).symm (f + f') = _
+  simp [AddEquivClass.map_add]
+
+@[simp]
+lemma homAddEquiv_symm_sub (X : C) (Y : D) (f f' : X ⟶ G.obj Y) :
+    (adj.homEquiv X Y).symm (f - f') = (adj.homEquiv X Y).symm f - (adj.homEquiv X Y).symm f' := by
+  change (adj.homAddEquiv X Y).symm (f - f') = _
+  simp [AddEquiv.map_sub]
+
+@[simp]
+lemma homAddEquiv_symm_neg (X : C) (Y : D) (f : X ⟶ G.obj Y) :
+    (adj.homEquiv X Y).symm (- f) = - (adj.homEquiv X Y).symm f := by
+  change (adj.homAddEquiv X Y).symm (- f) = _
+  simp [AddEquiv.map_neg]
+
+open Opposite in
+/-- If we have an adjunction `adj : F ⊣ G` of functors between preadditive categories,
+and if `F` is additive, then the hom set equivalence upgrades to an isomorphism between
+`G ⋙ preadditiveYoneda` and `preadditiveYoneda ⋙ F`, once we throw in the necessary
+universe lifting functors.
+Note that `F` is additive if and only if `G` is, by `Adjunction.right_adjoint_additive` and
+`Adjunction.left_adjoint_additive`.
+-/
+def compPreadditiveYonedaIso :
```

Comment:
I suggest defining the isomorphism in the other direction (the same as `addHomEquiv`):
```lean
def preadditiveYonedaCompIso :
    preadditiveYoneda ⋙ (whiskeringLeft _ _ _).obj F.op ⋙
      (whiskeringRight _ _ _).obj AddCommGrp.uliftFunctor.{max v₁ v₂} ≅
    G ⋙ preadditiveYoneda ⋙ (whiskeringRight _ _ _).obj AddCommGrp.uliftFunctor.{max v₁ v₂} :=
  NatIso.ofComponents
    (fun Y ↦ NatIso.ofComponents
      (fun X ↦ (AddEquiv.ulift.trans ((adj.homAddEquiv (unop X) Y).trans
        AddEquiv.ulift.symm)).toAddCommGrpIso)
      (fun g ↦ by
        ext ⟨y⟩
        exact AddEquiv.ulift.injective (adj.homEquiv_naturality_left g.unop y)))
    (fun f ↦ by
      ext _ ⟨x⟩
      exact AddEquiv.ulift.injective (adj.homEquiv_naturality_right x f))
```
(The simp lemmas for `preadditiveYoneda` are suboptimal...)

Reply:
I chose this direction to be consistent with [CategoryTheory.Adjunction.compYonedaIso](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/Adjunction/Basic.html#CategoryTheory.Adjunction.compYonedaIso), since I'm using almost the same name.

The `simp` lemmas for `preadditiveYoneda` often make the situation more complicated than it was to begin with.

Reply:
Whichever direction you use, you should try to clean up the proof similarly as above.